### PR TITLE
Introducing plugins and jasc_config variables 

### PR DESCRIPTION
**Pull request by attachmentgenie**
and adding small example

### CODE REVIEW

**Comment by attachmentgenie** [comment]
still trying to workout why this example crashes jenkins upon starting a worker in nomad and why [qa.nomad](https://github.com/attachmentgenie/nomad-jobs/blob/master/qa.nomad) doesnt

File: packs/jenkins/vars_cloud.nomad [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+plugins = ["configuration-as-code", "hashicorp-vault-plugin", "job-dsl", "nomad"]
+jasc_config = <<EOF
+jenkins:
+  agentProtocols:
+  - "JNLP4-connect"
+  - "Ping"
+  clouds:
+  - nomad:
+      name: "nomad"
+      nomadUrl: "http://{{ env "attr.unique.network.ip-address" }}:4646"
```
Comment (efbar) [inline packs/jenkins/vars_cloud.nomad:15]: here I would add the consul service variant, if the job is registering itself to consul.

File: packs/jenkins/vars_cloud.nomad [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                  },
+                  "Tasks": [
+                    {
+                      "Name": "jenkins-worker",
+                      "Driver": "docker",
+                      "Config": {
+                        "image": "jenkins/inbound-agent"
+                      },
+                      "Env": {
+                        "JENKINS_URL": "http://{{ env "NOMAD_ADDR_http" }}",
```
Comment (efbar) [inline packs/jenkins/vars_cloud.nomad:46]: here too.

**Comment by attachmentgenie** [comment]
@efbar good point again. i decided in the end to create a much simpler jasc example and will move the nomad cloud workers into a separate PR. It will take me a while to figure out what is going wrong, and i want this to move forward

**Comment by DerekStrickland** [comment]
Hi @attachmentgenie  

This looks like it has merge conflicts with the current main. Since you are closer to this code, do you mind taking a look and resolving?

**Review by DerekStrickland — CHANGES_REQUESTED** [review]
Merge conflicts detected. Sorry if I caused this by merging the chown PR.  Could you resolve and let me know? I'd love to get this in.

**Review by DerekStrickland — APPROVED** [review]
It's working great after the merge. Thanks so much for this contribution!

**Pull request merged**